deps: update collector to 0.125.0

dsotirakis opened the pull request:
Updates otel collector to `0.125.0`

Skipped `0.124.0` due to https://github.com/open-telemetry/opentelemetry-collector/issues/12907

guicaulada reviewed (commented):
I noticed lot's of formatting changes to workflows, and some code changes to .go files, are all these intended and part of the 0.125.0 update? 

I assume the `receivertest.NopType` is now a required argument for `receivertest.NewNopSettings` but other changes like the `generated_logs.go` feel like a new feature added to the receiver and `generated_metrics.go` feel like a logic change since the scope name will be dynamic now, if I understand correctly?

dsotirakis commented:
> I noticed lot's of formatting changes to workflows

Hm, I could only see single to double quote changes, which I reverted anyway in [d4b26d0](https://github.com/grafana/grafana-ci-otel-collector/pull/257/commits/d4b26d099d4822fa313df3a4804ba01a4df2df3e).

> but other changes like the generated_logs.go feel like a new feature added to the receiver

well this comes as part of [this issue](https://github.com/open-telemetry/opentelemetry-collector/issues/12571) which now comes as a default if logs are enabled.

> logic change since the scope name will be dynamic now

logic doesn't change, since what was in there before is now replaced by the var `ScopeName` which is hardcoded in `generated_status.go`.
